crates/matrix-sdk-appservice/src/lib.rs

```diff
@@ -709,25 +713,28 @@ impl AppService {
                     let key = &[USER_MEMBER, room_id.as_bytes(), b".", user_id.as_bytes()].concat();
                     let membership = match client.store().get_custom_value(key).await? {
                         Some(value) => String::from_utf8(value).ok().map(MembershipState::from),
+                        // Assume the appservice is in every known room
+                        None if user_id == appserv_uid => Some(MembershipState::Join),
```

Comment:
It seemed like a safe assumption that if the appservice receives events for a given room, the main client is already a member of it and we just missed it or haven't processed the relevant membership event yet.

Reply:
Yep, seems like a save assumption. A proper fix would be to force a full state sync if we see events for a room the first time, but that's out of scope for this PR.